transactions/transactor_test.go

```diff
@@ -313,6 +316,132 @@ func (s *TxQueueTestSuite) TestDiscardTransaction() {
 	s.NoError(WaitClosed(w, time.Second))
 }
 
+
+func (s *TxQueueTestSuite) TestTransactionsWithSimulatedBackend() {
+
+	testCases := []struct {
+		name             string
+		getTransactions  func(testaddr common.Address) []SendTxArgs
+		receiptValidator func(receipt *types.Receipt, testaddr common.Address)
+	}{
+		{
+			"EthTransfer",
+			func(testaddr common.Address) []SendTxArgs {
+				return []SendTxArgs{
+					{
+						Value: (*hexutil.Big)(big.NewInt(10)),
+					},
+				}
+			},
+			func(receipt *types.Receipt, testaddr common.Address) {
+				successStatus := uint(1)
+				s.Equal(successStatus, receipt.Status)
+			},
+		},
+		{
+			"ContractCreation",
+			func(testaddr common.Address) []SendTxArgs {
+				return []SendTxArgs{
+					{
+						Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+					},
+				}
+			},
+			func(receipt *types.Receipt, testaddr common.Address) {
+				s.Equal(crypto.CreateAddress(testaddr, 0), receipt.ContractAddress)
+			},
+		},
+		{
+			"ContractStateTransfer",
+			func(testaddr common.Address) []SendTxArgs {
+				// method ID: web3.utils.sha3('setOwner(bytes32,address)').substring(0, 8)
+				setOwnerMethodID := "0x5b0fc9"
+				node := make([]byte, 32)
+				ensParsed, _ := abi.JSON(strings.NewReader(contract.ENSABI))
+				setOwnerMethodData, _ := ensParsed.Pack("setOwner", setOwnerMethodID, node, testaddr)
+
+				// method ID: web3.utils.sha3('owner(bytes32)').substring(0, 8)
+				ownerMethodID := "0x02571b"
+				getOwnerMethodData, _ := ensParsed.Pack("owner", ownerMethodID, node)
+
+				return []SendTxArgs{
+					{
+						// Create contract
+						Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+					},
+					{
+						// Change contract owner
+						Input: hexutil.Bytes(setOwnerMethodData),
+					},
+					{
+						// Check that the contract owner is correct
+						Input: hexutil.Bytes(getOwnerMethodData),
+					},
+				}
+			},
+			func(receipt *types.Receipt, testaddr common.Address) {
+				successStatus := uint(1)
+				s.Equal(successStatus, receipt.Status)
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+
+			key, _ := crypto.GenerateKey()
+			testaddr := crypto.PubkeyToAddress(key.PublicKey)
+			genesis := core.GenesisAlloc{
+				testaddr: {Balance: big.NewInt(100000000000)},
+			}
+			backend := backends.NewSimulatedBackend(genesis)
+			selectedAccount := &account.SelectedExtKey{
+				Address:    testaddr,
+				AccountKey: &keystore.Key{PrivateKey: key},
+			}
+
+			s.manager.sender = backend
+			s.manager.gasCalculator = backend
+			s.manager.pendingNonceProvider = backend
+
+			transactions := testCase.getTransactions(testaddr)
+
+			go func() {
+				transactionsApproved := 0
+				for i := 1000; i > 0; i-- {
```

Comment:
this loop doesn't make any sense. i see that you didnt add it. but i it is weird and wrong, why to approve 1000 txs? and this is a source of those warnings about wrong key for approval